pkg/table/context/table.go

```diff
@@ -130,6 +130,19 @@ type MutateContext interface {
 	// which is a buffer for table related structures that aims to reuse memory and
 	// saves allocation.
 	GetMutateBuffers() *MutateBuffers
+
+	// SetExtraIndexKeyPosInfo sets the extra index key pos info.
+	// It's used when the given row is not a full row, but only contains some limit columns
+	SetExtraIndexKeyPosInfo(indexKeyPos map[int64][]int)
+
+	// GetExtraIndexKeyPosInfo gets the extra index key pos info.
+	GetExtraIndexKeyPosInfo(idxID int64) []int
```

Comment:
I don't think it's a good way to put these "temporary" state to `MutateContext` for reasons:

1. In the current implementation, the `MutateContext` is kept in the session instead of building it in place and discarded after use. That means it can be shared across scenes and it's better to expose some stable facts only to make it easy to use, such as system variable, etc.
2. I think `HasExtraInfo` and `GetExtraIndexKeyPosInfo` is only used to describe the input datums for `RemoveRecord`, right? So it's better to keep them together. `MutateContext` is also used by `AddRecord` and `UpdateRecord` and other scenes like lightning import or ddl reorging. These new methods will make the context confusing.

Why not to introduce a new struct to replace `[]types.Datum` as the input of `RemoveRecord`, for example:

```
type PartialRow struct {
    // d may contains the part of column datums of a row
    d []types.Datum
    // offsetMapping has a same size with d, it indicates the offset mapping the offset in `d` to the real offset
    offset mapping []int
}
```

So we can change signature the method `RemoveRecord` to:

```
RemoveRecord(ctx MutateContext, txn kv.Transaction, h kv.Handle, row PartialRow) error
```

And `FetchValues`

```
FetchValues(row PartialRow, columns []types.Datum) ([]types.Datum, error)
```

Reply:
Another way is to introduce `DeleteRecordOption` like `AddRecord` or `UpdateRecord`, if the input is a partial row, we can invoke `RemoveRecord` like this:

```
t.RemoveRecord(ctx.GetTableCtx(), txn, h, data, table.WithRowIndexMapping(posInfo.IndexesForDelete))
```